Add backlog.md and graphics to specs

msdisme opened the pull request:
Adding backlog.md to policy docs, making specs/assets directory. adding graphics to assets directory.

msdisme committed:
Writing up policy for backlog management

Initial policy doc for backlog and sprint backlog management.

Closes CCI-MOC/ops-issues#21

larsks commented:
Could you base this on the [policy template](https://github.com/CCI-MOC/ops-docs/blob/master/specs/policy-template.md)?

larsks committed:
fix additional html-ization of policy by michael's editor

larsks reviewed (commented):
It looks like `githubbacklog.png` is unused.

larsks committed:
fix image links

image links should be relative to the document (e.g.,
"assets/myimage.png"), and large images should probably just be linked
rather than embedded in the document.

msdisme committed:
Delete gihtubbacklog.png

This file is no longer referenced in backlog.md and is not needed in the assets directory.

msdisme commented:
>It looks like githubbacklog.png is unused.
Yes - I've deleted it from  `msdisme:master`

rob-baron reviewed (approved):
Look Ok.

naved001 commented on specs/backlog.md:
```diff
@@ -0,0 +1,92 @@
+## Alternatives & History
+
+Other approaches:
+1. Use Github Kanban model and do not have separate Scrum Boards for each sprint.   <--- **Honestly given our size we should probably shelve sprints and do Kannan.  What does team need to see to decide?** --->
```
typo: `"...shelve sprints and do Kannan.."` should be Kanban

msdisme replied:
Resolved by removing comment about using Kanban based on team feedback that they much prefer a scrum mode.

larsks commented on specs/backlog.md:
```diff
@@ -0,0 +1,92 @@
+# Creation of Backlog as a Ranked List
+
+We want to ensure that decisions about what to work on incorporate prioritization's agreed by the the team, product owners, our partners and customers. Specifically, we want to reach the point that:
```
`prioritization's` should be plural, not possessive.

msdisme replied:
resolved in doc

larsks commented on specs/backlog.md:
```diff
@@ -0,0 +1,92 @@
+	- this should be visible by a Rank Order List and, where appropriate, an explanation of the prioritization (example below).
+- The teams priorities are visible to interested parties. 
+- Items "ready to be worked on" state is easily identifiable.
+- The tool/process we use has a connection to GitHub.
```
Do you want to elaborate here on what is implied by "a connection to GitHub"?  Do you mean "the tool/process we use should react automatically to changes in Github issues and pull requests"?

msdisme replied:
Resolved in doc.

larsks commented on specs/backlog.md:
```diff
@@ -0,0 +1,92 @@
+
+To ensure that level of order and flexibility issues should be written as close to the code they relate to as possible and also tracked in the backlog for the development team.  
+
+Github supports this model inherently in their [Kanban model](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/about-project-boards) and, with a little bit of work can support it for scrum as well.  <--- we will need to validate by usage that the overhead is not onerous.  Also, a discussion of [Kanban or Scrum](https://www.mountaingoatsoftware.com/blog/when-kanb ...
```
Do you want to make your `<--- ... --->` comments less comment-y? Might as well just make them regular parts of the document.

msdisme replied:
Leaving for now as it will either be gone or lead to doc changes once we experience 1-3 sprints.

larsks commented on specs/backlog.md:
```diff
@@ -0,0 +1,92 @@
+
+Github supports this model inherently in their [Kanban model](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/about-project-boards) and, with a little bit of work can support it for scrum as well.  <--- we will need to validate by usage that the overhead is not onerous.  Also, a discussion of [Kanban or Scrum](https://www.mountaingoatsoftware.com/blog/when-kanb ...
+
+1. Issues should be written associated with the repository closest to the section of relevant code.  For example a hil/bmi issues would be written in CCI-MOC/hil. If no repository is appropriate they may be added directly to the [mocbacklog](https://github.com/orgs/CCI-MOC/projects/8).   
```
Maybe "issues should be created in the repository closest to the relevant code."

larsks replied:
"a hil/bmi issues" should be "a hil/bmi issue"

larsks replied:
I think "would be created in" instead of "would be written in" sounds more natural.

msdisme replied:
> Maybe "issues should be created in the repository closest to the relevant code."
> "a hil/bmi issues" should be "a hil/bmi issue"
> I think "would be created in" instead of "would be written in" sounds more natural.
Resolved all three issues in text.

larsks commented on specs/backlog.md:
```diff
@@ -0,0 +1,92 @@
+2. If you add the issue to the repository,whenever possible add the repository to [Linked Repositories](https://github.com/orgs/CCI-MOC/projects/8/settings/linked_repositories) under the Menu.  This way it will be included in default searches.  
+3. To find and order all cards available across all cci-moc repositories execute this search in the [add card](assets/githubaddcards.png) dialog:
+ 
+    `org:CCI-MOC state:open is:issue is:open repo:*`
```
You don't need the backtics here. If you indent code by four spaces, it will be formatted as a code sample. You can also fence the code like this if you prefer that to indentation:

    ```
    code sample goes here
    ```

msdisme replied:
resolvd in text.

larsks commented on specs/backlog.md:
```diff
@@ -0,0 +1,92 @@
+1. Open [mocbacklog](https://github.com/orgs/CCI-MOC/projects/8)
+2. Click + Add cards on the right side of the project board
+3. In the search field use this search: `org:CCI-MOC is:open is:issue repo:*`,  making sure "Only show results from linked repositories" is *not* checked.
+4. Drag the cards that show in the search results to the BackLog column add them <--- as a team figure out if we need two columns, one for order or "Ready to be worked" --->.
```
This sentence needs work. Maybe:

> Drag the cards that show in the search results to the BackLog column to add them to the backlog

msdisme replied:
I'd like to hold rewriting this till we actually try it on monday.

larsks commented on specs/backlog.md:
```diff
@@ -0,0 +1,92 @@
+### Scrum Setup
+1. Open [mocbacklog](https://github.com/orgs/CCI-MOC/projects/8) 
+2. Create a new Scrum Project with the name based on starting and end dates - eg. Sprint week 46 and 47 based on number for weeks of year.  
+3. In Sprint Planning meeting move items from Backlog to Sprint Project. 
```
Can an issue simply be "moved" from the backlog to the sprint project? Or is there more to the process than that? I thought we looked at that briefly on a call and it required a bit of a workaround, but maybe I'm misremembering.

msdisme replied:
This actually needs a full rewrite based on git board container and triage.    Will figure out if I can  do changes and request review.

msdisme replied:
Just realized, I haven't said thanks - thanks for reviewing, and the constructive feedback!

did some digging and this needs a rewrite.  Doing it here first then figuring out the best way to capture.  
3. In Sprint Planning Meeting items in [mocbacklog](https://github.com/orgs/CCI-MOC/projects/8) are added to a sprint project board following [these instructions](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/adding-issues-and-pull-requests-to-a-project-board#adding-issues-and-pull-requests-to-a-project-board-from-the-sidebar).

larsks commented on specs/backlog.md:
```diff
@@ -0,0 +1,92 @@
+
+### Milestones
+
+Policy will go into affect on Nov 16, 2020.
```
This is awfully optimistic. Possibly "Policy will go into affect when this proposal is accepted by the team".

msdisme replied:
Well, we're strating the sprint on the 16th, so I'll leave it for now :-)

msdisme committed:
Update backlog.md

Rewrote first paragraph to address Lars feedback: prioritization's should be plural, not possessive.

msdisme committed:
expand on github connection in line 10

expand on tool connection to githu in response to Lars feedback.

msdisme committed:
resolved comments in pull request for lines 38-40

> Maybe "issues should be created in the repository closest to the relevant code."
> "a hil/bmi issues" should be "a hil/bmi issue"
> I think "would be created in" instead of "would be written in" sounds more natural.

msdisme committed:
removed comment about kanban

both addressing comment about typo for Kanban and responding to feedback in friday standup that at this point the team has a preference for scrum.

larsks commented on specs/backlog.md:
```diff
@@ -31,11 +31,11 @@ The team has settled on GitHub Projects and Issues as the tool of choice. It mee
 The items in a backlog should be mutable both in their contents and their relative prioritization. Those changes should occur because of actions of the team, or team members, not due to side effects of the underlying tool.
 
-To ensure that level of order and flexibility issues should be written as close to the code they relate to as possible and also tracked in the backlog for the development team.  
+To ensure that level of order and flexibility issues should created in the repository closest to the relevant code and also tracked in the backlog for the development team.  
```
"should **be** created in ..."

msdisme committed:
removed unneccessary ` ('s)

removed unneccessary ` ('s)

msdisme commented on specs/backlog.md:
```diff
@@ -0,0 +1,92 @@
+### Scrum Setup
+1. Open [mocbacklog](https://github.com/orgs/CCI-MOC/projects/8) 
+2. Create a new Scrum Project with the name based on starting and end dates - eg. Sprint week 46 and 47 based on number for weeks of year.  
+3. In Sprint Planning meeting move items from Backlog to Sprint Project. 
```
Just realized, I haven't said thanks - thanks for reviewing, and the constructive feedback!

did some digging and this needs a rewrite.  Doing it here first then figuring out the best way to capture.  
3. In Sprint Planning Meeting items in [mocbacklog](https://github.com/orgs/CCI-MOC/projects/8) are added to a sprint project board following [these instructions](https://docs.github.com/en/free-pro-team@latest/github/managing-your-work-on-github/adding-issues-and-pull-requests-to-a-project-board#adding-issues-and-pull-requests-to-a-project-board-from-the-sidebar).

msdisme committed:
added missing "be" in line 34

msdisme committed:
Updated Scrum Setup section and moved scope up

in response to https://github.com/CCI-MOC/ops-docs/pull/5#r523080406 rewrote scrum setup section and moved scope from below scrum setup to above backlog setup/additions section.

knikolla reviewed (approved):
I do strongly agree with the direction of the proposed PR and I feel it clearly lays down the motivation for it. Some minor rewordings and clarifications (wrt methodology) can be implemented as a follow up after we have done a few sprints with this approach, and don't hold back my +2.

msdisme commented:
﻿Is the etiquette that with 2 approvers can push, or is it preferable for someone else to push ?

---
Michael Daitzman (He/Him)
msd@bu.edu
cell:978-201-9965



From: Kristi Nikolla <notifications@github.com>
Date: Wednesday, November 18, 2020 at 5:07 PM
To: CCI-MOC/ops-docs <ops-docs@noreply.github.com>
Cc: Daitzman, Michael S <msd@bu.edu>, Author <author@noreply.github.com>
Subject: Re: [CCI-MOC/ops-docs] Add backlog.md and graphics to specs (#5)

@knikolla approved this pull request.

I do strongly agree with the direction of the proposed PR and I feel it clearly lays down the motivation for it. Some minor rewordings and clarifications (wrt methodology) can be implemented as a follow up after we have done a few sprints with this approach, and don't hold back my +2.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/CCI-MOC/ops-docs/pull/5#pullrequestreview-533920224>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ABOXT4JDEYOH3GPFFINR2MDSQRAQJANCNFSM4TTME2HQ>.
